src/resources/elements/typefield.js

```diff
@@ -191,20 +191,22 @@ export class Typefield extends Field {
     if (this.keyKey) {
       value[this.keyKey] = this.key;
     }
-    if (this.showType) {
-      value.type = this.selectedType;
+    if (this.typeKey) {
+      value[this.typeKey] = this.selectedType;
     }
     return value;
   }
 
   setValue(value) {
     if (this.keyKey && value.hasOwnProperty(this.keyKey)) {
       this.key = value[this.keyKey];
+      delete value[this.keyKey];
     }
-    if (!this.showType) {
-      this.child.setValue(value);
-    } else if (typeof value === 'object' && !Array.isArray(value)) {
-      delete value.type;
+    if (this.typeKey && value.hasOwnProperty(this.typeKey)) {
```

Comment:
Add a descriptive comment to both of these blocks. English language can describe the goal of this `if` statement without the noisy JavaScript syntax.